core/src/main/scala/org/apache/spark/sql/rapids/tool/store/AccumInfo.scala

```diff
@@ -32,81 +32,133 @@ import org.apache.spark.sql.rapids.tool.util.EventUtils.parseAccumFieldToLong
  * @param infoRef - AccumMetaRef for the accumulator
  */
 class AccumInfo(val infoRef: AccumMetaRef) {
-  // TODO: use sorted maps for stageIDs and taskIds?
-  val taskUpdatesMap: mutable.HashMap[Long, Long] =
-    new mutable.HashMap[Long, Long]()
-  val stageValuesMap: mutable.HashMap[Int, Long] =
-    new mutable.HashMap[Int, Long]()
+  /**
+   * Maps stageId to a tuple of:
+   * - Total accumulator value for the stage (Long)
+   * - Statistical metrics for task-level updates (StatisticsMetrics)
+   */
+  val stageValuesMap: mutable.HashMap[Int, (Long, StatisticsMetrics)] =
+    new mutable.HashMap[Int, (Long, StatisticsMetrics)]()
 
   /**
-   * Add accumulable to a stage while:
-   * 1- handling StageCompleted event.
-   * 2- handling taskEnd event, we want to make sure that there is an entry that maps the stage to
-   *    the accumulable. The reason is that we want to keep track of the stageIDs in case their
-   *    stageCompleted has never been triggered. It is common case for incomplete eventlogs.
-   * @param stageId the stageId pulled from the StageCompleted/TaskEnd event
-   * @param accumulableInfo the accumulableInfo from the StageCompleted/TaskEnd event
-   * @param update optional long that represents the task update value in case the call was
-   *               triggered by a taskEnd event and the map between stage-Acc has not been
-   *               established yet.
+   * Adds or updates accumulator information for a stage.
+   * Called during StageCompleted or TaskEnd events processing.
+   *
+   * @param stageId The ID of the stage
+   * @param accumulableInfo Accumulator information from the event
+   * @param update Optional task update value for TaskEnd events
    */
   def addAccumToStage(stageId: Int,
       accumulableInfo: AccumulableInfo,
       update: Option[Long] = None): Unit = {
     val parsedValue = accumulableInfo.value.flatMap(parseAccumFieldToLong)
     // in case there is an out of order event, the value showing up later could be
     // lower-than the previous value. In that case we should take the maximum.
-    val existingValue = stageValuesMap.getOrElse(stageId, 0L)
+    val existingEntry = stageValuesMap.getOrElse(stageId,
+      (0L, StatisticsMetrics(0L, 0L, 0L, 0L)))
     val incomingValue = parsedValue match {
       case Some(v) => v
       case _ => update.getOrElse(0L)
     }
-    stageValuesMap.put(stageId, Math.max(existingValue, incomingValue))
+    val newValue = Math.max(existingEntry._1, incomingValue)
```

Comment:
add more code comments here to explain why we take the max and any other necessary details.